File: plugin/src/test/java/testcode/serial/SerialisationFalsePositive.java
package testcode.serial;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.Serializable;

public class SerialisationFalsePositive implements Serializable {
    int param1;
    Object param2;

    private void readObject(ObjectInputStream inputStream) throws IOException, ClassNotFoundException {
        param1 = inputStream.readInt();
        param2 = inputStream.readObject();
    }
}

File: findsecbugs-samples-kotlin/src/test/kotlin/com/h3xstream/findsecbugs/deserialisation/SerialisationFalsePositive.kt
package com.h3xstream.findsecbugs.deserialisation

import java.io.IOException
import java.io.ObjectInputStream
import java.io.Serializable

class SerialisationFalsePositive : Serializable {
    internal var param1: Int = 0
    private var param2: Any? = null

    @Throws(IOException::class, ClassNotFoundException::class)
    private fun readObject(inputStream: ObjectInputStream) {
        param1 = inputStream.readInt()
        param2 = inputStream.readObject()
    }
}